### ISSUE TITLE
ghc-7.8: Agda-2.5.2 depends on geniplate-mirror-0.7.4 which failed to install

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#2338]
There seems to be a new dependency (since yesterday) on geniplate-mirror-0.7.4 which does not build.
https://github.com/danr/geniplate/issues/6

**Comment by andreasabel** [comment]
@nad ?

**Comment by asr** [comment]
Removed the 2.5.2 milestone because the release will be based on the latest commit [successful](https://travis-ci.org/agda/agda/builds/183533331) on Travis.

**Comment by andreasabel** [comment]
Ok.

**Comment by asr** [comment]
Fixed by 17612b2.

**Comment by andreasabel** [comment]
After pulling your fix, I still get the same build error:
```
$ make install-bin
cabal install --disable-documentation --builddir=./dist-2.5.2 --enable-tests --disable-library-profiling --program-suffix=-2.5.2 
Resolving dependencies...
In order, the following will be installed:
geniplate-mirror-0.7.4 (via: Agda-2.5.2) (new version)
Agda-2.5.2 *test (reinstall) (changes: EdisonCore-1.3.1.1 added, QuickCheck-2.9.2 added, equivalence-0.3.1 added, filemanip-0.3.6.3 added, geniplate-mirror-0.7.1 removed, gitrev-1.2.0 added, pretty-1.1.1.1 -> 1.1.3.4, process-extras-0.4.1.4 added, regex-tdfa-text-1.0.0.3 added, semigroups-0.18.1 -> 0.18.2, tasty-0.11.0.4 added, tasty-silver-3.1.9 added, template-haskell-2.9.0.0 -> 2.11.0.0, temporary-1.2.0.4 added, transformers-compat-0.4.0.4 -> 0.5.1.4, void-0.7.1 added)
Warning: Note that reinstalls are always dangerous. Continuing anyway...
cabal: Entering directory '/tmp/cabal-tmp-32206/geniplate-mirror-0.7.4'
Configuring geniplate-mirror-0.7.4...
Building geniplate-mirror-0.7.4...
Preprocessing library geniplate-mirror-0.7.4...
[1 of 1] Compiling Data.Generics.Geniplate ( Data/Generics/Geniplate.hs, dist-2.5.2/build/Data/Generics/Geniplate.o )

Data/Generics/Geniplate.hs:84:25:
    Couldn't match expected type ‘Dec’ with actual type ‘[Dec] -> Dec’
    Probable cause: ‘InstanceD’ is applied to too few arguments
    In the expression:
      InstanceD [] (foldl AppT (ConT cls) ts) (funDef met e)
...
```
But I fixed it now by unregistering the `template-haskell-2.11.0.0` which probably came by a forced reinstallation of agda's dependencies.
